[8.2] API documentation for Endpoint Host isolation Exceptions (#1643)
elastic/security-docs: docs/management/api/host-isolation-exceptions-api.asciidoc

:endpoint-artifact-list-id:                     endpoint_host_isolation_exceptions
:endpoint-artifact-name:                        host isolation exceptions
:endpoint-artifact-name-single:                 host isolation exception
:endpoint-artifact-name-sentence-start:         Host isolation exceptions
:endpoint-artifact-name-single-sentence-start:  Host isolation exception
[[host-isolation-exceptions-api]]
=== {endpoint-artifact-name-sentence-start}

> Create, retrieve, update, and delete endpoint <<host-isolation-exceptions>> via API. {endpoint-artifact-name} are managed via the <<exceptions-api-overview>> using a static container id (`list_id`) of `pass:a[{endpoint-artifact-list-id}]`, which must be created prior to adding a trusted application. Access to these APIs requires that a user has authorization to manage endpoints.

Reviewer: Use the start of sentence attribute here + reference the attribute for artifact type:

```suggestion
Create, retrieve, update, and delete endpoint <<host-isolation-exceptions>> via API. {endpoint-artifact-name-sentence-start} are managed via the <<exceptions-api-overview>> using a static container id (`list_id`) of `pass:a[{endpoint-artifact-list-id}]`, which must be created prior to adding the {endpoint-artifact-name}. Access to these APIs requires that a user has authorization to manage endpoints.
```

--------------------------------------------------
{
	"_version": "WzIxNDcsMV0=",
	"created_at": "2022-03-03T14:04:00.496Z",
	"created_by": "elastic",
	"description": "some description about this entry",
	"entries": [
		{
			"field": "destination.ip",
			"value": "10.10.0.1",
			"type": "match",
			"operator": "included"
		}
	],
	"id": "c6883700-9afa-11ec-94b1-fd7e90cc2a08",
	"item_id": "1342081d-66ad-4b03-b554-131195ce404b",
	"list_id": "endpoint_host_isolation_exceptions",
	"name": "Host Isolation Exception - Created with API",
	"namespace_type": "agnostic",
	"os_types": [
		"linux",
		"macos",
		"windows"
	],
	"tags": [
		"policy:all"
	],
	"tie_breaker_id": "b65843e1-07b8-46d2-9e2b-5f0586dc364a",
	"type": "simple",
	"updated_at": "2022-03-03T14:04:00.498Z",
	"updated_by": "elastic"
}
--------------------------------------------------

[#update-item-api]
==== Update {endpoint-artifact-name-single}

`PUT <kibana host>:<port>/api/exception_lists/items`

===== Request body

A JSON object containing the fields listed below. The request must have the following:

include::_exceptions-api-reusable-content.asciidoc[tags=endpoint-exceptions-request-payload-requirements]

> 

Reviewer: same here:

```suggestion
* `os_types` must be an array with all xref:exception-item-os-types[]
```

> * `os_types` must be an array with all xref:exception-item-os-types[]

Reviewer 2: Rephrase to match other items in list. Also, I think the `*` at the beginning of the line is causing it to render as a sub-bullet for some reason; changing to `-` might fix it.
```suggestion
- The `os_types` value must be an array with all xref:exception-item-os-types[].
```
<img width="500" alt="image" src="https://user-images.githubusercontent.com/87339667/156829399-9efc11ee-e0f7-4086-920f-a9e30cbd4bcd.png">

// KIBANA

===== Response code

`200`::
    Indicates a successful call.

===== Response payload

[source,json,subs="attributes"]
--------------------------------------------------
{
	"_version": "WzIyMDIsMV0=",
	"created_at": "2022-03-03T14:04:00.496Z",
	"created_by": "elastic",
	"description": "via API",
	"entries": [
		{
			"field": "destination.ip",
			"type": "match",
			"value": "10.10.0.15",
			"operator": "included"
		}
	],
	"id": "c6883700-9afa-11ec-94b1-fd7e90cc2a08",
	"item_id": "1342081d-66ad-4b03-b554-131195ce404b",
	"list_id": "endpoint_host_isolation_exceptions",
	"name": "HIE via API",
	"namespace_type": "agnostic",
	"os_types": [
		"linux",
		"macos",
		"windows"
	],
	"tags": [],
	"tie_breaker_id": "b65843e1-07b8-46d2-9e2b-5f0586dc364a",
	"type": "simple",
	"updated_at": "2022-03-03T14:22:08.345Z",
	"updated_by": "elastic"
}
--------------------------------------------------

==== Delete {endpoint-artifact-name-single}

`DELETE <kibana host>:<port>/api/exception_lists/items`

===== URL query parameters

include::_exceptions-api-reusable-content.asciidoc[tags=api-id-or-itemid-query-params]

====== Example request

> Deletes a {endpoint-artifact-name-single} with `item_id` of `1342081d-66ad-4b03-b554-131195ce404b`:

Reviewer 2: Revise to avoid conflict with "a/an" in other artifact name (like "event filter").
```suggestion
Deletes the {endpoint-artifact-name-single} with `item_id` of `1342081d-66ad-4b03-b554-131195ce404b`:
```

{
  "data": [
    {
      "_version": "WzEzNjcsMV0=",
      "created_at": "2022-03-01T16:24:39.471Z",
      "created_by": "elastic",
      "description": "Host Isolation Exception item via API",
      "entries": [
        {
          "field": "destination.ip",
          "type": "match",
          "value": "10.10.0.15",
          "operator": "included"
        }
      ],
      "id": "4e0e4e10-9af8-11ec-94b1-fd7e90cc2a08",
      "item_id": "6555a6ed-b31c-4c74-b1a1-f2c5d9fb7d4a",
      "list_id": "{endpoint-artifact-list-id}",
      "name": "Host Isolation Exception via API",
      "namespace_type": "agnostic",
      "os_types": [
        "linux",
        "macos",
        "windows"
      ],
      "tags": [
        "policy:all"
      ],
      "tie_breaker_id": "97fe9d23-c8b8-4dc1-97e1-16444c8d863a",
      "type": "simple",
      "updated_at": "2022-03-01T17:21:07.273Z",
      "updated_by": "elastic"
    }
  ],
  "page": 1,
  "per_page": 10,
  "total": 1
}
--------------------------------------------------

include::_exceptions-api-reusable-content.asciidoc[tag=endpoint-exceptions-value-types-info-section]

[#exception-item-entry-schema]
===== `entry` object schema

> {endpoint-artifact-name} allow for at most 1 condition to be defined. The following entry is supported by {endpoint-artifact-name}:

Reviewer: Use start of sentence attribute:

```suggestion
{endpoint-artifact-name-sentence-start} allow for at most 1 condition to be defined. The following entry is supported by {endpoint-artifact-name}:
```

====== Destination IPs

Destination IPs are supported by all xref:exception-item-os-types[OS types]. A destination IP entry has the following structure:

[source,json]
-----------------------------
{
  "field": "destination.ip",
  "value": "192.168.1.1",
  "type": "match",
  "operator": "included"
}
-----------------------------

> * `field` : `destination.ip`
> * `value` : a valid IPv4 address with optional CIDR associated with the `field`
> * `type` : must be `match`
> * `operator` : must be `included`

Reviewer 2: Minor reformatting for consistency.
```suggestion
* `field` : `destination.ip`.
* `value` : A valid IPv4 address with optional CIDR associated with the `field`.
* `type` : Must be `match`.
* `operator` : Must be `included`.
```

:endpoint-artifact-list-id:                     endpoint_host_isolation_exceptions
:endpoint-artifact-name:                        host isolation exceptions
:endpoint-artifact-name-single:                 host isolation exception
:endpoint-artifact-name-sentence-start:         Host isolation exceptions
:endpoint-artifact-name-single-sentence-start:  Host isolation exception
:endpoint-artifact-api-doc-name:                host-isolation-exception

[[host-isolation-exceptions-api]]
=== {endpoint-artifact-name-sentence-start}

> Create, retrieve, update, and delete endpoint <<host-isolation-exceptions>> via API. {endpoint-artifact-name-sentence-start} are managed via the <<exceptions-api-overview>> using a static container id (`list_id`) of `pass:a[{endpoint-artifact-list-id}]`, which must be created prior to adding the {endpoint-artifact-name}. Access to these APIs requires that a user has authorization to manage endpoints.

Reviewer 3: ```suggestion
Create, retrieve, update, and delete <<host-isolation-exceptions, host isolation exceptions>> via API. {endpoint-artifact-name-sentence-start} are managed via the <<exceptions-api-overview>> using a static container id (`list_id`) of `pass:a[{endpoint-artifact-list-id}]`, which must be created prior to adding the {endpoint-artifact-name}. Access to these APIs requires that a user has authorization to manage endpoints.
```
